@Tags decorator not working in 6.5.0

## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Context (Environment)

Version of the library: 6.5.0
Version of NodeJS: 18.19.0

- Confirm you were using yarn not npm: [x]


---

6.5.0 seem to have broke for me the @Tags decorator.

This code has been working fine on 6.4.0:

```
import { Controller, Get, Response, Route, Security, Tags } from 'tsoa';

import * as M from '../../../models';
import * as R from '../../../utils/response';
import InfoService from './service';

@Route('v1/info')
@Tags('Informations')
@Security('BearerAuth')
@Response<R.UnauthorizedError>(
	401,
	'Unauthorized - Please add BearerAuth to your header',
	R.unauthorizedError,
)
export class InfoController extends Controller {
	private service: InfoService;

	constructor() {
		super();
		this.service = new InfoService();
	}

	@Get('/')
	public async getInfos(): Promise<R.full<M.Info.InfoResponse, unknown>> {
		const info = await this.service.getInfos();
		return R.fullResponse<M.Info.InfoResponse, unknown>(
			true,
			'Informations successfully fetched',
			info,
		);
	}
}
```

After updating to 6.5.0 I am getting those typescript errors:

```
src/components/v1/z_info/controller.ts:8:1 - error TS1238: Unable to resolve signature of class decorator when called as an expression.
  The runtime will invoke the decorator with 1 arguments, but the decorator expects 3.

@Tags('Informations')
  ~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.decorators.legacy.d.ts:21:52
    21 declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'propertyKey' was not provided.
src/components/v1/z_info/controller.ts:8:2 - error TS1270: Decorator function return type 'void | TypedPropertyDescriptor<unknown>' is not assignable to type 'void | typeof InfoController'.
  Type 'TypedPropertyDescriptor<unknown>' is not assignable to type 'void | typeof InfoController'.

@Tags('Informations')
   ~~~~~~~~~~~~~~~~~~~~
```

No other changes to used deps in package.json has been made. It was working until I just updated tsoa to 6.5.0.
But in case you need my other used packages:

```
	"dependencies": {
		"axios": "^1.7.7",
		"axios-retry": "^4.5.0",
		"cloudconvert": "^2.3.7",
		"compression": "^1.7.4",
		"deepl-node": "^1.14.0",
		"dotenv": "^16.4.5",
		"drizzle-dbml-generator": "^0.9.0",
		"drizzle-kit": "^0.25.0",
		"drizzle-orm": "^0.34.1",
		"envalid": "^8.0.0",
		"express": "^5.0.0",
		"express-rate-limit": "^7.4.1",
		"helmet": "^8.0.0",
		"mime-types": "^2.1.35",
		"module-alias": "^2.2.3",
		"morgan": "^1.10.0",
		"multer": "^1.4.5-lts.1",
		"postgres": "^3.4.4",
		"request-ip": "^3.3.0",
		"sharp": "^0.33.5",
		"swagger-ui-express": "^5.0.1",
		"tsoa": "^6.5.0",
		"tsx": "^4.19.1",
		"winston": "^3.15.0",
		"winston-daily-rotate-file": "^5.0.0",
		"zod": "^3.23.8"
	},
	"devDependencies": {
		"@types/compression": "^1.7.5",
		"@types/express": "^5.0.0",
		"@types/mime-types": "^2.1.4",
		"@types/module-alias": "^2.0.4",
		"@types/morgan": "^1.9.9",
		"@types/multer": "^1.4.12",
		"@types/node": "^22.7.5",
		"@types/request-ip": "^0.0.41",
		"@types/swagger-ui-express": "^4.1.6",
		"@typescript-eslint/eslint-plugin": "^8.8.1",
		"@typescript-eslint/parser": "^8.8.1",
		"depcheck": "^1.4.7",
		"eslint": "^9.12.0",
		"eslint-config-prettier": "^9.1.0",
		"eslint-plugin-prettier": "^5.2.1",
		"eslint-plugin-simple-import-sort": "^12.1.1",
		"globals": "^15.11.0",
		"husky": "^9.1.6",
		"nodemon": "^3.1.7",
		"prettier": "^3.3.3",
		"ts-node": "^10.9.2",
		"typescript": "^5.6.3"
	},
```

Screenshot just for better visualization.

![grafik](https://github.com/user-attachments/assets/74b371ea-6013-4964-8c4f-c6e03459358f)

Update: See my first comment for more context.

@WoH

--- a/packages/runtime/src/decorators/tags.ts
+++ b/packages/runtime/src/decorators/tags.ts
@@ -1,4 +1,4 @@
-export function Tags(...values: string[]): MethodDecorator {
+export function Tags(...values: string[]): ClassDecorator & MethodDecorator {
   return () => {
     return;
   };

--- a/tests/fixtures/controllers/methodController.ts
+++ b/tests/fixtures/controllers/methodController.ts
@@ -30,6 +30,7 @@ const TEST_SEC = {
 const ATT_KEY9 = 'x-attKey9';
 
 @Route('MethodTest')
+@Tags('MethodTest')
 export class MethodController extends Controller {
   @Options('Options')
   public async optionsMethod(): Promise<TestModel> {

--- a/tests/unit/swagger/definitionsGeneration/metadata.spec.ts
+++ b/tests/unit/swagger/definitionsGeneration/metadata.spec.ts
@@ -155,7 +155,7 @@ describe('Metadata generation', () => {
         throw new Error('Method tags not defined!');
       }
 
-      expect(method.tags).to.deep.equal(['Tag1', 'Tag2', 'Tag3']);
+      expect(method.tags).to.deep.equal(['Tag1', 'Tag2', 'Tag3', 'MethodTest']);
     });
 
     it('should generate multi response', () => {
@@ -227,7 +227,7 @@ describe('Metadata generation', () => {
         secondSec: ['permission:admin', 'permission:owner'],
       });
 
-      expect(method.tags).to.deep.equal(['EnumTag1']);
+      expect(method.tags).to.deep.equal(['EnumTag1', 'MethodTest']);
     });
 
     it('should generate success response', () => {

